src/main/java/org/opensearch/sdk/handlers/ExtensionsRestRequestHandler.java

```diff
@@ -32,13 +36,15 @@
 public class ExtensionsRestRequestHandler {
     private static final Logger logger = LogManager.getLogger(ExtensionsRestRequestHandler.class);
     private final ExtensionRestPathRegistry extensionRestPathRegistry;
+    private NamedXContentRegistry namedXContentRegistry;
 
     /**
      * Instantiate this class with an existing registry
      *
      * @param restPathRegistry The ExtensionsRunnerer's REST path registry
      */
-    public ExtensionsRestRequestHandler(ExtensionRestPathRegistry restPathRegistry) {
+    public ExtensionsRestRequestHandler(ExtensionRestPathRegistry restPathRegistry, SDKNamedXContentRegistry sdkNamedXContentRegistry) {
+        this.namedXContentRegistry = sdkNamedXContentRegistry.getRegistry();
```

Comment:
(Similar to above, don't call `getRegistry()` here on initialization, call it when it's needed in the handler method.